### PR TITLE
chore(auth): add license, README.md and e2e sample

### PR DESCRIPTION
**Pull request by samisayegh**
https://coveord.atlassian.net/browse/KIT-1239

### CODE REVIEW

File: packages/auth/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# @coveo/auth
```
Comment (samisayegh) [inline packages/auth/README.md:1]: Preview: https://github.com/coveo/ui-kit/tree/KIT-1239-d/packages/auth

File: packages/auth/rollup.config.js [hunk 2]
```diff
@@ -1,17 +1,43 @@
...
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+  `
+  .replace('[yyyy]', new Date().getFullYear())
+  .replace('[name of copyright owner]', 'Coveo Solutions Inc.')
+
+  return licensePlugin({banner});
+}
```
Comment (samisayegh) [inline packages/auth/rollup.config.js:28]: Will need to do a similar thing for other projects in ui-kit. We are not implementing the license properly 🤞

File: packages/auth/samples/saml/index.html [hunk 3]
```diff
@@ -0,0 +1,22 @@
+<html>
+  <head>
+    <script type="module">
+      import {buildSamlClient} from '../../src/auth.ts';
+
+      // Test values for "organizationId and "provider" can be generated using:
+      // https://github.com/coveo/searchfuncttests/blob/develop/searchapi/auth/samlOktaMultipleProviderWithQuery.ts
```
Comment (samisayegh) [inline packages/auth/samples/saml/index.html:7]: This comment is probably not enough ... there some setup involved to run that test file including setting up env variables and creating an Okta account.

Maybe I'll create a confluence doc on Friday ... atm the steps are part of the slack lore.
@acote-coveo, @dreisch-coveo do we have a doc by any chance? If not any preferences on where to create one?

Comment (acote-coveo) [reply packages/auth/samples/saml/index.html:7]: I think if we need further documentation on how to run those test it should be in the README. I can add more details to it tomorrow

File: packages/auth/README.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+Functions to help authenticate with the Coveo platform.
+
+## Install
+
+```
+npm i @coveo/saml
+```
+
+## SAML
```
Comment (olamothe) [inline packages/auth/README.md:11]: I would add a `SAML with @coveo/headless` section/sample

File: packages/auth/samples/saml/index.html (hunk 3 above)
Comment (olamothe) [reply packages/auth/samples/saml/index.html:7]: Is it really important that people can run this ? What's the value in that ?

Isn't this the important documentation ?  https://docs.coveo.com/en/91/build-a-search-ui/saml-authentication

I think we should also write an article in docs.coveo.com/headless > Usage > SAML

**Review by olamothe — APPROVED** [review]

**Review by btaillon — APPROVED** [review]

File: packages/auth/samples/saml/index.html (hunk 3 above)
Comment (acote-coveo) [reply packages/auth/samples/saml/index.html:7]: Who's the targeted audience for `samples` in Headless? It people outside from Coveo have access to that, I don't see how relevant it is to share a private repo link ^^

Comment (samisayegh) [reply packages/auth/samples/saml/index.html:7]: I had this file in mind more for e2e manual testing vs. a public audience. I think I'll rename `samples` -> `manual-e2e`.

An automated e2e test would be nice too, though there some set up involved.

**Review by ThibodeauJF — APPROVED** [review]

**Pull request merged**